pollard_test.go

```diff
@@ -1289,3 +1290,76 @@ func compareNodeMap(mapA, mapB map[miniHash]*polNode) error {
 
 	return fmt.Errorf(str)
 }
+
+// Test that the cached nodes are correct.
+func TestProofCacheNodes(t *testing.T) {
+
+	// Create a new accumulator with 10 leaves.
+	adds := make([]Leaf, 10)
+	for i := range adds {
+		adds[i] = Leaf{Hash: sha256.Sum256([]byte{uint8(i)})}
+	}
+
+	// Create the accumulator and add one leaf at a time.
+	p := NewAccumulator(true)
+	for _, leaf := range adds {
+		err := p.Modify([]Leaf{leaf}, nil, Proof{})
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	fmt.Printf("Merkle tree:\n\n%s\n", p.String())
+
+	// Check that the nodes are cached correctly.
+	for i := range adds {
+		fmt.Printf("For node at position: %x\n", i)
+		leaves, _, err := p.getNodesForProof(uint64(i))
+		if err != nil {
+			fmt.Print("hre")
+			fmt.Printf("error: %x\n", i)
+			t.Fatal(err)
+		}
+
+		// Leaf hashes that should be cached ideally
+		cachedLeaves := []Leaf{}
+		for _, leaf := range leaves {
+			if leaf.Remember {
```

Comment:
You haven't marked any leaves to be remembered when you `leaves`. This branch will never execute